xbout/plotting/utils.py

```diff
@@ -31,3 +34,387 @@ def plot_separatrix(da, sep_pos, ax, radial_coord='x'):
+        # plot lower inner leg
+        region1 = da[:, ystart:(j11 + 1)]
+
+        yind = [j11, j22 + 1]
+        region2 = da[:ix1, yind]
+
+        region3 = da[ix1:, j11: (j11 + 2)]
+
+        yind = [j22, j11 + 1]
+        region4 = da[:ix1, yind]
+
+        regions.extend([region1, region2, region3, region4])
+
+        ystart = j11 + 1
+
+    if j21 + 1 > ystart:
+        # Inner SOL
+        region5 = da[:, ystart:(j21 + 1)]
+        regions.append(region5)
+
+        ystart = j21 + 1
+
+    if j12 > j21:
+        # Contains upper PF region
+
+        # Inner leg
+        region6 = da[ix1:, j21:(j21 + 2)]
+        region7 = da[:, ystart:nin]
+
+        # Outer leg
+        region8 = da[:, nin:(j12 + 1)]
+        region9 = da[ix1:, j12:(j12 + 2)]
+
+        yind = [j21, j12 + 1]
+
+        region10 = da[:ix1, yind]
+
+        yind = [j21 + 1, j12]
+        region11 = da[:ix1, yind]
+
+        regions.extend([region6, region7, region8,
+                        region9, region10, region11])
+
+        ystart = j12 + 1
+    else:
+        ystart -= 1
+
+    if j22 + 1 > ystart:
+        # Outer SOL
+        region12 = da[:, ystart:(j22 + 1)]
+        regions.append(region12)
+
+        ystart = j22 + 1
+
+    if j22 + 1 < ny:
+        # Outer leg
+        region13 = da[ix1:, j22:(j22 + 2)]
+        region14 = da[:, ystart:ny]
+
+        # X-point regions
+        corner1 = da[ix1 - 1, j11]
+        corner2 = da[ix1, j11]
+        corner3 = da[ix1, j11 + 1]
+        corner4 = da[ix1 - 1, j11 + 1]
+
+        xregion_lower = xr.concat([corner1, corner2, corner3, corner4],
+                                  dim='dim1')
+
+        corner5 = da[ix1 - 1, j22 + 1]
+        corner6 = da[ix1, j22 + 1]
+        corner7 = da[ix1, j22]
+        corner8 = da[ix1 - 1, j22]
+
+        xregion_upper = xr.concat([corner5, corner6, corner7, corner8],
+                                  dim='dim1')
+
+        region15 = xr.concat([xregion_lower, xregion_upper], dim='dim2')
+
+        regions.extend([region13, region14, region15])
+
+    if j21 > j11 and j12 > j21 and j22 > j12:
+        # X-point regions
+        corner1 = da[ix1-1, j12]
+        corner2 = da[ix1, j12]
+        corner3 = da[ix1, j12+1]
+        corner4 = da[ix1-1, j12+1]
+
+        xregion_lower = xr.concat([corner1, corner2, corner3, corner4],
+                                  dim='dim1')
+
+        corner5 = da[ix1 - 1, j21 + 1]
+        corner6 = da[ix1, j21+1]
+        corner7 = da[ix1, j21]
+        corner8 = da[ix1 - 1, j21]
+
+        xregion_upper = xr.concat([corner5, corner6, corner7, corner8],
+                                  dim='dim1')
+
+        region16 = xr.concat([xregion_lower, xregion_upper], dim='dim2')
+
+        regions.append(region16)
+
+    return regions
+
+
+def plot_separatrices(da, ax):
+    """Plot separatrices"""
+
+    j11, j12, j21, j22, ix1, ix2, nin, nx, ny, y_boundary_guards = _get_seps(da)
+
+    R = da.coords['R'].transpose('x', 'theta')
```

Comment:
The rest of this function's logic assumes that `R` and `Z` are numpy arrays, so this should have `.values()` here.

It would be possible to rewrite the separatrix-plotting logic to use `xarray.DataArray`s instead but I'm not sure if we would gain anything really.